python/shotgun_fields/shotgun_field_delegate.py

```diff
@@ -45,36 +46,81 @@ def __init__(self, sg_entity_type, field_name, display_class, editor_class, view
         self._editor_class = editor_class
         self._bg_task_manager = bg_task_manager
 
+    def paint(self, painter, style_options, model_index):
+        """
+        Paint method to handle all cells that are not being currently edited.
+
+        :param painter:         The painter instance to use when painting
+        :param style_options:   The style options to use when painting
+        :param model_index:     The index in the data model that needs to be painted
+        """
+
+        # let the base class do all the heavy lifting
+        super(ShotgunFieldDelegate, self).paint(painter, style_options, model_index)
+
+        # clear out the paint widget's contents to prevent it from showing in
+        # other places in the view (since the widget is shared)
+        widget = self._get_painter_widget(model_index, self.view)
+        widget.set_value(None)
+
     def _create_widget(self, parent):
         """
-        :param parent:  QWidget to parent the widget to
-        :type parent:   :class:`~PySide.QtGui.QWidget`
+        :param parent: QWidget to parent the widget to
+        :type parent: :class:`~PySide.QtGui.QWidget`
 
-        :returns:       QWidget that will be used to paint grid cells in the view.
-        :rtype:         :class:`~PySide.QtGui.QWidget`
+        :returns: QWidget that will be used to paint grid cells in the view.
+        :rtype: :class:`~PySide.QtGui.QWidget`
         """
         widget = self._display_class(
             parent=parent,
             entity_type=self._entity_type,
             field_name=self._field_name,
             bg_task_manager=self._bg_task_manager,
+            delegate=True,
         )
+
+        if self._display_class == self._editor_class:
+            # display and edit classes are the same. we need to make sure
+            # we disable the editing
+            widget.enable_editing(False)
```

Comment:
Yes, having the same class for both is possible and encouraged. We only really want the display class here but this call ensures that editing is disabled so that the delegate isn't drawing the widget in its edit state. That make sense? I updated the comment too to make that clearer. :)